lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -778,14 +803,51 @@ def accessible_objects(user_or_attribute, operation, options = {})
       end
     end
 
+    # Returns a map of operation names to the user_or_attribute's accessible objects via each operation
+    def accessible_objects_for_operations(user_or_attribute, operations, options = {})
+      unless options[:direct_only]
+        raise ArgumentError, 'Functionality for indirect objects is not yet implemented!'
+      end
+
+      # convert to operation names if operation instances given
+      operation_names = operations.map { |o| operation_identifier(o) }
+
+      # default objects for each operation to empty list
+      accessible_map = operation_names.map { |o| [o, []] }.to_h
+
+      if options[:ignore_prohibitions]
+        accessible_map.merge!(
+          objects_for_user_or_attribute_and_operations(user_or_attribute, operation_names, options)
+        )
+        return accessible_map
+      end
+
+      # using `prohibitions_for` rather than just mapping to `prohibition_identifier`
+      # because we want to confirm that these prohibitions exist in the db
+      prohibition_names = prohibitions_for(operation_names).map { |p| operation_identifier(p) }
+      op_and_prohib_names = operation_names + prohibition_names
+      accessible_map.merge!(
+        objects_for_user_or_attribute_and_operations(user_or_attribute, op_and_prohib_names, options)
+      )
+
+      operation_names.each do |operation_name|
+        prohibition_name = prohibition_identifier(operation_name)
+        prohibited_objects = accessible_map[prohibition_name] || []
+        accessible_map[operation_name] -= prohibited_objects
+        accessible_map.delete(prohibition_name)
```

Comment:
i might be reading this wrong but i think with this approach if i have `read` on `dog` and `frog` and `~read` on `dog` the accessible map will be: 
```
{ 'read' => ['frog'] }
```
i think there's a problem with this (similar to an observation you made previously for priv#report stuff) because a user can be assigned on an all collection but prohibited on a single operable. so if i had `read` on `all_frogs` and `~read` on `tree_frog` this fn as is would return
```
{ 'read' => ['all_frogs'] }
```
meaning the information on the individual prohibition is lost.

i think we can get around this by just letting the caller of the function (dalton) sort it out, so having this function return:
for the first example 
```
{ 'read' => ['dog', 'frog'], '~read' => ['dog'] }
```
and for the second 
```
{ 'read' => ['all_frogs'], '~read' => ['tree_frog'] }
```

Reply:
ah yes right the plan is to call this with `operations + prohibitions` 👍